ops/model.py

```diff
@@ -60,6 +62,16 @@ def get_unit(self, unit_name):
     def get_app(self, app_name):
         return self._cache.get(Application, app_name)
 
+    @property
+    def endpoints(self):
```

Comment:
I find "endpoints" a bit confusing since we also use that term when talking about relations.  Maybe "bindings" or "network_endpoints" (though that's unfortunately long).  Just "networks" is misleading and "spaces" is inaccurate.

Reply:
I always thought of `binding`s as tuples like `(endpoint_name, space)` so it is more an operation rather than an object.

https://jaas.ai/docs/spaces#heading--using-spaces
`A binding is a space-specific, software level operation...`

At the same time, endpoints in the charm metadata are simply strings while in this PR I endpoints are identified by a string + an optional relation ID.

I could see `binding`s as objects holding references both a relation and network info.

- relation.binding -> reference to a `Binding` object;
    - `relation.binding.ingress_address`;
- binding.relation -> weakref to a `Relation` object if present.

The problem is that there are `extra-bindings` which don't have a relation. So the weakref back to a relation is questionable and having `EndpointMapping` would be needed to support `extra-bindings` while `relation.binding` could be there for convenience.

Let's discuss it during the stand-up.